controllers/user.js

```diff
@@ -27,24 +28,110 @@ router.get( '/:id', (req, res, next) => {
 
 
 router.put( '/:id', (req, res, next) => {
-  req.user.update( req.body, (error, result) => {
+  // mask data (only pass on qualifying values)
+  const updateData = {};
+  if ( req.body.level ) updateData.level = req.body.level;
+  if ( req.body.newbie === false || req.body.newbie === true ) updateData.newbie = req.body.newbie;
+  if ( req.body.name ) updateData.name = req.body.name;
+
+  req.user.update( updateData, (error, result) => {
     if ( error ) return next(error);
     res.json(result);
   });
 });
 
 
 
-router.get( '/:id/lexemes',
-  (req, res, next) => {
-    // get list of IDs of user's words
-    User.findById( req.user._id )
-      .then( user => {
-        const ids = user.lexemes.map( lexeme => lexeme.lexeme_id )
-        console.log(ids);
-      })
-      .catch( error => next(error) );
+router.get( '/:id/lexemes', (req, res, next) => {
+  User.findById( req.user._id ) // get again b/c in req.user the lexemes are stripped off
+    .populate('lexemes.lexeme', 'lexeme')
+    .then( ({ lexemes }) => res.json(lexemes) )
+    .catch( error => next(error) );
+});
+
+
+router.put( '/:id/lexemes', (req, res, next) => {
+
+  // move to models.js?
+  User.findById( req.user._id )
+    .then( ({ lexemes: oldLexemes }) => {
+
+      // remove
+      const lexemesFiltered = oldLexemes
+        .filter( ({ lexeme: lexemeId}) => req.body.remove.indexOf(String(lexemeId)) === -1 );
+
+      // add
+      const lexemesToAdd = req.body.add
+        .filter( lexemeId => oldLexemes.map( ({lexeme: id}) => String(id) ).indexOf(lexemeId) === -1 ) // filter out if already in lexeme list
+        .map( lexeme => { return {lexeme} } );
+
+      // persist in DB, deliver response
+      req.user.update( { lexemes: [...lexemesFiltered, ...lexemesToAdd] }, (error, result) => {
+        if ( error ) return next(error);
+        res.json(result); // just "res.status(200).send()" ?
+      });
+
+    })
+    .catch( error => next(error) );
+});
+
+
+router.get( '/:id/lexemes/:quantity', (req, res, next) => {
+  User.findById( req.user._id )
+    .populate('lexemes.lexeme')
+    .then( ({ lexemes }) => {
+      // select questions, later based on (1) level of control, (2) last learned
+      return res.json(lexemes);
+    })
+    .catch( error => next(error) );
+});
+
+
+router.put( '/:id/lexemes/:lexemeId/:success', (req, res, next) => {
+  User.findById( req.user._id )
+    .then( user => {
+      const lexeme = user.lexemes.filter( ({ lexeme: id }) => String(id) === req.params.lexemeId )[0];
```

Comment:
[find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) suits better than filter because you only expect one.